extensions/amp-carousel/0.1/test/test-slides.js

```diff
@@ -750,6 +750,70 @@ describe('Slides functional', () => {
       expect(slides[index]).to.equal('e');
     });
   });
+
+  describe('Navigating slides', () => {
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+      setupElements();
+      setupSlides();
+      slides.inViewport_ = true;
+      slides.getVsync = function() {
+        return {
+          mutate: function(fn) {
+            fn();
+          },
+        };
+      };
+      slides.deferMutate = function(fn) {
+        fn();
+      };
+      slides.scheduleLayout = sandbox.spy();
+      slides.updateInViewport = sandbox.spy();
+      slides.schedulePause = sandbox.spy();
+      slides.schedulePreload = sandbox.spy();
+    });
+    afterEach(() => {
+      sandbox.restore();
+      teardownElements();
+    });
+
+    it('should hide slides that are not the current one', () => {
+      expect(slide0.style.visibility).to.be.equal('visible');
+      expect(slide1.style.visibility).to.be.equal('hidden');
+      expect(slide2.style.visibility).to.be.equal('hidden');
+
+      slides.goCallback(1, /*animate*/ false);
+      expect(slide0.style.visibility).to.be.equal('hidden');
+      expect(slide1.style.visibility).to.be.equal('visible');
+      expect(slide2.style.visibility).to.be.equal('hidden');
+
+      slides.goCallback(-1, /*animate*/ false);
+      expect(slide0.style.visibility).to.be.equal('visible');
+      expect(slide1.style.visibility).to.be.equal('hidden');
+      expect(slide2.style.visibility).to.be.equal('hidden');
+    });
+
+    it('should preload slides in direction of navigation', () => {
+      expect(slide0.style.visibility).to.be.equal('visible');
+      expect(slide1.style.visibility).to.be.equal('hidden');
+      expect(slides.scheduleLayout.calledWith(slide0));
```

Comment:
Actually I think these are not running properly. These need to `.to.be.true` appended, will update.